File: app/src/main/java/com/example/ge62/diplomahotelreserv/ui/HotDealsActivity.java
//package com.example.ge62.diplomahotelreserv.ui;
//
//import android.content.Intent;
//import android.os.Bundle;
//import android.support.v7.app.AppCompatActivity;
//import android.support.v7.widget.LinearLayoutManager;
//import android.support.v7.widget.RecyclerView;
//import android.support.v7.widget.Toolbar;
//import android.view.View;
//
//import com.example.ge62.diplomahotelreserv.R;
//import com.example.ge62.diplomahotelreserv.adapters.HotDealsEditListAdapter;
//import com.example.ge62.diplomahotelreserv.adapters.HotHotelsCollection;
//import com.example.ge62.diplomahotelreserv.adapters.RVClickListner;
//import com.example.ge62.diplomahotelreserv.models.HotDealHotels;
//
//import java.util.ArrayList;
//import java.util.List;
//
//public class HotDealsActivity extends AppCompatActivity {
//
//    private List<HotDealHotels> hotelsList = new ArrayList<>();
//    private RecyclerView recyclerView;
//    private HotDealsEditListAdapter hotDealsListAdapter;
//    HotDealHotels hottel;
//    private RVClickListner <HotDealHotels>rvClickListner = new RVClickListner<HotDealHotels>() {
//        @Override
//        public void onItemClick(HotDealHotels item, int target) {
//            Intent i=new Intent(HotDealsActivity.this,FullItemInfoHotDealsActivity.class);
//            i.putExtra("hottel",item);
//            startActivity(i);
//        }
//    };
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_hot_deals);
//        recyclerView = (RecyclerView) findViewById(R.id.hotHotellsList);
//        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
//        recyclerView.setLayoutManager(new LinearLayoutManager(this));
//        //hotDealsListAdapter = new HotDealsEditListAdapter(this, HotHotelsCollection.getHottels());
//        recyclerView.setAdapter(new HotDealsEditListAdapter(this,HotHotelsCollection.getHottels(),rvClickListner));
//
//        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                onBackPressed();
//            }
//        });
////        recyclerView = (RecyclerView) findViewById(R.id.hotHotellsList);
////
////        hotDealsListAdapter = new HotDealsEditListAdapter(hotelsList);
////        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getApplicationContext());
////        recyclerView.setLayoutManager(mLayoutManager);
////        recyclerView.setItemAnimator(new DefaultItemAnimator());
////        recyclerView.setAdapter(hotDealsListAdapter);
////
////        prepareHotHotellData();
////    }
////
////    private void prepareHotHotellData() {
////        HotDealHotels hotel= new HotDealHotels(R.drawable.hotel1, "test", "wadsadasd");
////        hotelsList.add(hotel);
////
////
////        hotDealsListAdapter.notifyDataSetChanged();
////    }
//    }
//
//
//}

File: app/src/main/java/com/example/ge62/diplomahotelreserv/ui/hotdeals/HotDealsActivity.kt
package com.example.ge62.diplomahotelreserv.ui.hotdeals

import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.example.ge62.diplomahotelreserv.BaseActivity
import com.example.ge62.diplomahotelreserv.BookingApp
import com.example.ge62.diplomahotelreserv.adapters.HotDealsListAdapter
import com.example.ge62.diplomahotelreserv.adapters.RVClickListner
import com.example.ge62.diplomahotelreserv.models.Hotel
import com.example.ge62.diplomahotelreserv.rest.NetworkManager
import com.example.ge62.diplomahotelreserv.rest.SmartNetworkCallback
import kotlinx.android.synthetic.main.activity_hot_deals.*
import javax.inject.Inject



class HotDealsActivity : BaseActivity() {

    override fun getLayoutRes(): Int = com.example.ge62.diplomahotelreserv.R.layout.activity_hot_deals

    @Inject
    lateinit var networkManager: NetworkManager



    private var listHottels = arrayListOf<Hotel>()
    init {
        BookingApp.getAppComponent().inject(this)
    }

    override fun initViews() {
       initToolbar(toolbar," ",true)




        //     recyclerView?.setAdapter(HotDealsEditListAdapter(this, ,ho, rvClickListner))
        //recyclerView?.setAdapter(HotDealsEditListAdapter(this, HotHotelsCollection.getHottels(), rvClickListner))

    }
    private var recyclerView: RecyclerView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        recyclerView = findViewById(com.example.ge62.diplomahotelreserv.R.id.hotHotellsList) as RecyclerView
        recyclerView?.setLayoutManager(LinearLayoutManager(this))

        networkManager.getAllHottels(object : SmartNetworkCallback<List<Hotel>>() {
            override fun onResult(data: List<Hotel>) {
                super.onResult(data)
                listHottels.clear()
                listHottels.addAll(data)
                initRv()
            //    updateProjectsList()
            }
        })


    }

    private fun initRv() {
        recyclerView?.setAdapter(HotDealsListAdapter(this,listHottels , rvClickListner))
    }

    private val rvClickListner = RVClickListner<Hotel> { item, target ->
        var i = Intent(this@HotDealsActivity, FullItemInfoHotDealsActivity::class.java)
        //i.putExtra("hottel", item)
//        var lat =item.latitude!!.toDouble()
//        var long = item.longitude!!.toDouble()
        i.putExtra("description",item.description)
        i.putExtra("hotelName",item.hottelName)
        i.putExtra("regionName",item.regionName)
      //  i.putExtra("latitude",item.latitude)
     //   i.putExtra("longitude",item.longitude)
        i.putExtra("phone",item.phone)
        i.putExtra("email",item.email)
        i.putExtra("image",item.image)

        val b = Bundle()
        b.putDouble("lat", item.latitude!!)
        b.putDouble("long",item.longitude!!)
        i.putExtras(b)
        startActivity(i)
    }
    override fun release() {
    }

}





